File: JumpGame/JumpGame.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        if len(nums) == 0:
            return False
        maxstep = nums[0]
        for i in range(1, len(nums)):
            if maxstep > 0:
                maxstep -= 1
                maxstep = max(maxstep, nums[i])
            else:
                return False
        return True

File: JumpGame/JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() == 0)
        	return false;
        int maxstep = nums[0];
        for(int i = 1; i < nums.size(); i++) {
        	if(maxstep > 0) {
        		maxstep--;
        		maxstep = max(maxstep, nums[i]);
        	}
        	else
        		return false;
        }
        return true;
    }
};